ambry-api/src/main/java/com.github.ambry/router/CryptoService.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import java.nio.ByteBuffer;
+
+
+/**
+ * Crypto Service to assist in encrypting or decrypting bytes
+ * T refers to the key type that this {@link CryptoService} accepts.
+ * Ensure that {@link CryptoService} implementation is compatible with the key type that
+ * {@link KeyManagementService} generates
+ */
+public interface CryptoService<T> {
```

Comment:
shouldn't this be `Closeable`? Won't there be resources to release during shutdown?